### ISSUE TITLE
base64: Missing details and man page for `openssl base64`; weird silent decoding failures on missing newline

### ISSUE THREAD

**Issue by DDvO** [openssl/openssl#18780]
I was asked by a colleague which of the base64 variants defined in https://en.wikipedia.org/wiki/Base64#Variants_summary_table are supported by OpenSSL.

There is `openssl base64` but no specific man page such as `doc/man1/openssl-base64.pod.in` for it -
likely this is because the command is somehow an alias to `openssl enc`, though this is not obvious.

There is some API-level documentation in `doc/man3/BIO_f_base64.pod`,
but it does not say which variants according to the various RFCs are supported.
Can anyone tell?

**Comment by DDvO** [comment]
The above man3 page says:
```
For reading, first line should be at most 1024  characters  long.
If it is longer then it is ignored completely.  Other input lines
can be of any length. There must be a newline at the end of input.
```
yet the newline at input end is not strictly needed.
OTOH, if a newline is not among the first 1024 characters, decoding **fails silently**.

**Comment by DDvO** [comment]
Here is a workaround for this weird behavior: prepend a newline to the input to be decoded.

**Comment by DDvO** [comment]
> Here is a workaround for this weird behavior: prepend a newline to the input to be decoded.

A simpler workaround is to set `BIO_FLAGS_BASE64_NO_NL` (i.e., use the `-A` option with `openssl base64`).

**Comment by paulidale** [comment]
I'd call this a bug not a documentation problem.

**Comment by t8m** [comment]
> I'd call this a bug not a documentation problem.

I disagree. It makes sense to be able to detect whether something is at least somewhat  formatted BASE64 (i.e. with lines split) or not.

**Comment by DDvO** [comment]
Yes, but it should not fail silently, pretending that an empty input was provided.
I suspect even more bugs in that ugly stretch if code.

**Comment by DDvO** [comment]
Another discrepancy with the docs was that on problem with the first line, two lines are skipped rather than one.
I found this experimenting with base64 input having a first line with 1024+ characters.

**Comment by t8m** [comment]
Please note we are looking for community help implementing this issue. Please comment if you are interested in implementing this feature or if you have reasons why this feature is important enough so we should consider implementing it with higher priority than other requested features. The issue will be closed after 2 months otherwise.

**Comment by t8m** [comment]
This issue was closed because it was marked with the help wanted label and there was no interest in implementing this issue or any other activity for more than 2 months.
